src/protocol.rs

```diff
@@ -199,6 +223,25 @@ impl<P, S, Z> From<Protocol<Rec<P>, S, Z>> for Protocol<P, S, Z> {
     }
 }
 
+impl<P, S, Z> fmt::Debug for Protocol<P, S, Z> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Protocol")
+            .field("id", &self.id)
+            .field("node_id", &self.node_id)
+            .field("tag", &self.tag)
+            .field("protocol", &any::type_name::<P>())
+            .field("serializer", &any::type_name::<S>())
+            .field("Z", &any::type_name::<Z>())
```

Comment:
I'm not sure what to call `Z` here.. any ideas? Or should I just remove this type from being debugged?